src/CameraServer.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+ * This file is part of the Camera Streaming Daemon
+ *
+ * Copyright (C) 2017  Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <assert.h>
+#include <cstring>
+#include <set>
+#include <sstream>
+#include <string.h>
+
+#include "CameraComponent_V4L2.h"
+#include "CameraServer.h"
+#include "log.h"
+#include "mavlink_server.h"
+#include "util.h"
+
+#define DEVICE_PATH "/dev/"
+#define VIDEO_PREFIX "video"
+#define DEFAULT_SERVICE_PORT 8554
+
+CameraServer::CameraServer(ConfFile &conf)
+    : mavlink_server(conf, streams, rtsp_server)
+    , rtsp_server(streams, DEFAULT_SERVICE_PORT)
+    , cameraCount(0)
+{
+    cameraCount = detectCamera(conf);
+
+    std::vector<CameraComponent *>::iterator it;
+    for (it = cameraList.begin(); it != cameraList.end(); it++) {
+        if (*it) {
+            mavlink_server.addCameraComponent(*it);
+        }
+    }
+}
+
+CameraServer::~CameraServer()
+{
+    stop();
+}
+
+void CameraServer::start()
+{
+    log_error("CAMERA SERVER START");
+
+    mavlink_server.start();
+}
+
+void CameraServer::stop()
+{
+    mavlink_server.stop();
+}
+
+// prepare the list of cameras in the system
+int CameraServer::detectCamera(ConfFile &conf)
+{
+    int count = 0;
+
+    count += detect_v4l2devices(conf, cameraList);
+
+    return count;
+}
+
+static void _insert(std::set<std::string> *value_set, const char *val, size_t val_len)
+{
+    while (isspace(*val) && val_len > 0) {
+        val++;
+        val_len--;
+    }
+
+    while (val_len > 0 && isspace(val[val_len - 1]))
+        val_len--;
+
+    if (val_len)
+        value_set->insert(std::string(val, val_len));
+}
+
+static int parse_stl_set(const char *val, size_t val_len, void *storage, size_t storage_len)
```

Comment:
I will check if this function can be moved to ConfFile class and can be used by both classes